stl/inc/xmemory

```diff
@@ -1746,6 +1746,12 @@ _Alloc_ptr_t<_Alloc> _Uninitialized_fill_n(
         _Fill_memset(_Unfancy(_First), _Val, static_cast<size_t>(_Count));
         return _First + _Count;
     } else {
+        if constexpr (_Fill_zero_memset_is_safe<_Ty*, _Ty> && _Uses_default_construct<_Alloc, _Ty*, _Ty>::value) {
+            if (_Is_scalar_zero(_Val)) {

```

Comment:
Ditto "Does this new special case have test coverage?"

Reply:
I found only one place where it's tested and it only tests non-default allocator with custom `construct` so it doesn't even apply for the already existing `memset` optimization.
https://github.com/microsoft/STL/blob/12c684bba78f9b032050526abdebf14f58ca26a3/tests/std/tests/Dev09_153419_tr1_allocators/test.cpp#L416-L421 Should I introduce more test coverage for this function?

Reply:
This function is used to implement `vector` resizing; I think simply initializing a `vector<void*>(43, nullptr)` will call it.

Reply:
Done